packages/yourloops/components/team/member-row.tsx

```diff
@@ -0,0 +1,140 @@
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import React, { useState } from "react";
+
+import { makeStyles, Theme } from "@material-ui/core/styles";
+import AccessTimeIcon from "@material-ui/icons/AccessTime";
+
+import { Team, TeamMember, useTeam } from "../../lib/team";
+import { StyledTableCell } from "../../pages/hcp/patients/table";
+import { UserInvitationStatus } from "../../models/generic";
+import Checkbox from "@material-ui/core/Checkbox";
+import { TeamMemberRole } from "../../models/team";
+import { useAuth } from "../../lib/auth";
+import { StyledTableRow } from "./common";
+
+const useStyles = makeStyles((theme: Theme) => ({
+  icon: {
+    width: "56px",
+    alignItems: "center",
+    justifyContent: "center",
+  },
+  iconCell: {
+    alignItems: "center",
+    display: "flex",
+    width: "67px",
+    justifyContent: "center",
+  },
+  tableRow: {
+    height: "55px",
+  },
+  typography: {
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    whiteSpace: "nowrap",
+    color: theme.palette.common.black,
+  },
+}));
+
+export interface TeamMembersProps {
+  team: Team;
+  teamMember: TeamMember;
+}
+
+function MemberRow(props: TeamMembersProps): JSX.Element {
+  const { teamMember, team } = props;
+  const classes = useStyles();
+  const rowId = teamMember.user.userid.replace(/@/g, "_");
+  const teamHook = useTeam();
+  const authContext = useAuth();
+  const [userUpdateInProgress, setUserUpdateInProgress] = useState<boolean>(false);
+  const currentUserId = teamMember.user.userid;
+  const loggedInUserId = authContext.user?.userid as string;
+  const loggedInUserIsAdmin = teamHook.isUserAdministrator(team, loggedInUserId);
+  const currentUserIsAdmin = teamHook.isUserAdministrator(team, currentUserId);
+  const currentUserIsPending = teamMember.status === UserInvitationStatus.pending;
+  const checkboxAdminDisabled = !loggedInUserIsAdmin || currentUserIsPending
+    || (loggedInUserId === currentUserId && teamHook.isUserTheOnlyAdministrator(team, loggedInUserId))
+    || userUpdateInProgress;
+
+  const switchRole = async (event: React.ChangeEvent<HTMLInputElement>): Promise<void> => {
+    const isAdmin = event.target.checked;
+    setUserUpdateInProgress(true);
+    await teamHook.changeMemberRole(teamMember, isAdmin ? TeamMemberRole.admin : TeamMemberRole.member);
+    setUserUpdateInProgress(false);
+  };
```

Comment:
When I switch role the disabled property of the checkbox is not refreshed.

Use case :
* team with one admin + one member (not admin)
* logged in as the admin user
* the admin checkbox is disabled for the admin connected member (because i'm the only admin)
* Promoting the not admin member to admin through the checkbox work
* the disabled checkbox should be now enabled (there are two admins now) -> it remains disabled

Another corner case when I invite a new member as an admin and he has not yet accepted the invite, I can remove myself the admin role... The number of admins should be filtered out of pending invites.